### DISCUSSION TITLE
Tanya seputar react

### DISCUSSION THREAD

**Discussion by deeninsom** [deaafrizal/forum#3] [Troubleshooting Room] [answered]
Misi bang, kenalin aku syihab, mahasiswa yang tidak pingin lulus cepat 🤣 dan memiliki ketertarikan didunia IT, sekarang lagi ngerjain beberapa joki web, sama jadi freelance backend (karena udah keluar dari perusahan yang lama 😅), aku ada pertanyaan nih bang, aku selama nyambi jadi front end, masih belum sebegitu paham tentang struktur folder yang rapih di frontend khususnya di react, barangkali bang dea ada waktu luang nih, bisa jawab pertanyaan ku 😅, thanks bang sehat selalu 🚀

**Comment by deaafrizal** [comment]
react itu dibuat dengan flexibilitasnya bro, jadi mau lu atur folder kek gimana juga kan sabi sabi aja. nah kalau gue pribadi biasanya setup di awal bikin folder structure ciri khas gue & langsung di diskusiin sama team. sebelum di diskusiin gue buat docs dari apa yang udah di setup. mulai dari penjelasan tiap folder, kenapa komponent ada disini, kenapa libs ada disana & aturan main penamaan file, fungsi, variabel. yang jelas gimana cara biar si team bisa kerja dengan nyaman tanpa pusing jauh harus mikirin folder ini utk apa itu untuk apa.

singkatnya kalau gak mau ambil pusing setup ala lu sendiri, lu bisa cari referensi best practice nya react developer. kalau gue biasanya comot beberepa konsepnya terus gue modif sesuai kualitas, pemahaman & skill tim yang kerja bareng gue.

contoh yg rada aneh aja ada yg kek gini di salah 1 projekan gue
![image](https://github.com/deaaprizal/forum/assets/43339356/8cdf9edd-cd0a-47c7-9993-4270972359e4)

apakah ini salah? tentu tidak, dari docs resmi nya si framework / library yg kita pake aja gak pernah mengecam developernya dalam menentukan structure. tapi again balik lagi, ada yang namanya *best practice* dan boleh lu pake, ataupun lu mau pake aturan lu sendiri. selama lu gak pusing dengan projekan lu & hasilnya bagus, dont make it hard 👌

**Reply by deaafrizal** [reply, marked as answer]
sebagai tambahan referensi.

structure grouping based on tipe file:
![image](https://github.com/deaaprizal/forum/assets/43339356/af960a94-b4c4-4336-9502-6344fe3280f4)

based on fitur:
![image](https://github.com/deaaprizal/forum/assets/43339356/54a71859-206e-4bb0-b857-18c06a1360fa)

dan terakhir, dari docs officialnya react sendiri bilang gini:
![image](https://github.com/deaaprizal/forum/assets/43339356/49668d49-3e85-4315-9f8c-627929766227)

**Reply by deeninsom** [reply]
wah gila sih, ngga expect dijelasin se detail ini wkwk, thanks bang pencerahanya  😅🙏, intinya brati yang penting kita ngga bingung sendiri ketika lagi develop 😅

**Comment by qwinza** [comment]
hallo bg, kenalin saya Faris, saya mau nanya kalo pake zustand di react tuh statenya dalam kondisi gimana yaa bg dea, masih bingung penggunaanya hehe

**Reply by deaafrizal** [reply]
coba bikin thread baru aja bro di forum [troubleshooting room](https://github.com/deaaprizal/forum/discussions/categories/troubleshooting-room) kasih hashtag react-js

biar dapet jawaban dari temen2 yang laen juga

**Reply by qwinza** [reply]
okee siap bg, hatur nuhun
